fabtests/unit/mr_cache_evict.c

```diff
@@ -0,0 +1,688 @@
+/*
+ * (C) Copyright 2020 Hewlett Packard Enterprise Development LP
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <unistd.h>
+#include <stddef.h>
+#include <sys/mman.h>
+#include <getopt.h>
+#include <fcntl.h>
+#include <time.h>
+#include <limits.h>
+
+#ifdef __linux__
+#include <malloc.h>
+#endif
+
+#include "unit_common.h"
+#include "shared.h"
+
+/* Supported memory region types. */
+enum mem_type {
+	MMAP,
+	BRK,
+	SBRK,
+};
+
+struct memory_region {
+	enum mem_type type;
+	size_t size;
+	char data[0];
+};
```

Comment:
Now that I can see what's going on better, I would change data back to char *data.  Have the alloc/free calls allocated and free data only, rather than struct memory_region.  struct memory_region should be declared as a global or stack variable.  Otherwise, if the user specifies a size less than sizeof(memory_region) the program will crash.

Reply:
Will do.

Reply:
In addition, I think I can get rid of struct memory_region completely since the size of the buffer is a global variable.